src/lib.rs

```diff
@@ -101,25 +101,25 @@ pub fn set_open_files_limit(mut _new_limit: isize) -> bool {
     cfg_if::cfg_if! {
         if #[cfg(all(not(feature = "unknown-ci"), any(target_os = "linux", target_os = "android")))]
         {
+            use crate::unix::linux::system::REMAINING_FILES;
+            use std::sync::atomic::Ordering;
+
             if _new_limit < 0 {
                 _new_limit = 0;
             }
             let max = sys::system::get_max_nb_fds();
             if _new_limit > max {
                 _new_limit = max;
             }
-            unsafe {
-                if let Ok(ref mut x) = sys::system::REMAINING_FILES.lock() {
-                    // If files are already open, to be sure that the number won't be bigger when those
-                    // files are closed, we subtract the current number of opened files to the new
-                    // limit.
-                    let diff = max.saturating_sub(**x);
-                    **x = _new_limit.saturating_sub(diff);
-                    true
-                } else {
-                    false
-                }
-            }
+
+            // If files are already open, to be sure that the number won't be bigger when those
+            // files are closed, we subtract the current number of opened files to the new
+            // limit.
+            let remaining = REMAINING_FILES.load(Ordering::SeqCst);
+            let diff = max.saturating_sub(remaining);
+            REMAINING_FILES.store(_new_limit.saturating_sub(diff), Ordering::SeqCst);
```

Comment:
Didn't use atomics much so my concerns might only come from my lack of understanding. In here, it's possible (even if unlikely) that the value changes between the call to `load` and the call to `store`, right? Isn't it possible to make this modification in one step instead?

Reply:
I think there could have been that issue, but I've fixed it by using fetch_update.